src/app/dataValidator/helpers/validators/validateType.js

```diff
@@ -1,13 +1,31 @@
 const { log, throwError } = require('../../utilities/messaging');
 
+const validateTypeOf = (schemaType, dataNode, parentSchemaName) => {
+  if (schemaType === `${typeof dataNode}`) {
+    log(`- Valid type of ${typeof dataNode}`);
+  } else {
+    throwError(
+      `Error: Type does not match for '${parentSchemaName}' - expected '${schemaType}' got '${typeof dataNode}'`,
+    );
+  }
+};
+
+const validateArray = (dataNode, parentSchemaName) => {
+  if (Array.isArray(dataNode)) {
+    log('- Valid type of array');
+  } else {
+    throwError(
+      `Error: Type does not match for '${parentSchemaName}' - expected 'array'`,
```

Comment:
Worth making this match line 8?

```suggestion
      `Error: Type does not match for '${parentSchemaName}' - expected 'array' got '${typeof dataNode}'`,
```

Reply:
This could be misleading because `typeof` on an Array returns `object`. So if there was a regression where this an `array` slipped through the if statement some how it would log `... - expected 'array' got 'object'` even though it is an array.

Reply:
Oh ok - so for example if the if statement got accidentally changed to `if (!Array.isArray(dataNode)) {` (note the `!`), then if we passed a legitimate array we'd fail the `if` and get the `else`, and it would say `... - expected 'array' got 'object'`

Seems fair. Maybe change to simply `got ${dataNode}`?

Reply:
`dataNode` in this case will pretty much always log `[object Object]` but quickly google'd and I think `console.log(JSON.stringify(dataNode))` would give us what we want